tests/test_render.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+from django.shortcuts import render
+from django.template import Context, RequestContext
+from django.template.response import SimpleTemplateResponse, TemplateResponse
+from django.test import RequestFactory
+
+
+class TestUserFacingTemplateRenderSuccess:
+    """
+    Test that the user-facing rendering entrypoints work correctly with dict contexts.
+    """
+
+    def test_render_shortcut_with_dict(self, template_engine):
+        """
+        `render()` shortcut function
+
+        See https://docs.djangoproject.com/en/stable/topics/http/shortcuts/#render
+        """
+        request = RequestFactory().get("/")
+        template = "basic.txt"
+        context = {"user": "Lily"}
+
+        response = render(request, template, context, using=template_engine.name)
+        assert response.status_code == 200
+        assert response.content == b"Hello Lily!\n"
+
+    def test_template_response_rendered_content(self, template_engine):
+        """
+        `TemplateResponse` -- a "lazy" way to render a template with a request context
+
+        This is used in django.views.generic.TemplateView
+
+        See https://docs.djangoproject.com/en/stable/ref/template-response/#templateresponse-objects
+        """
+        request = RequestFactory().get("/")
+        template = template_engine.from_string("{{ foo }}")
+        context = {"foo": "bar"}
+
+        response = TemplateResponse(request, template, context)
```

Comment:
It would be nice if we had a test depending on the `request` for the rendered content, since that's the main difference between `TemplateResponse` and `SimpleTemplateResponse`.

That could be this test or a new one (or maybe one already exists?).